modules/stanford_icon_grid/stanford_icon_grid.module

```diff
@@ -0,0 +1,220 @@
+      ),
+    ),
+    '#default_value' => $items[$delta]['icon'],
+  );
+  return array('icon' => $element + $icon);
+}
+
+/**
+ * Gets available font awesome icons from the json file.
+ *
+ * @return array
+ *   Font awesome options in groups.
+ */
+function stanford_icon_grid_get_icons() {
+  // If data is cached, return the data.
+  if ($cache = cache_get('stanford_icon_grid_icons')) {
+    return $cache->data;
+  }
+
+  // Get json data and set cache.
+  $fa_icons = drupal_get_path('module', 'stanford_icon_grid') . '/css/font-awesome.json';
+  $fa_icons = drupal_json_decode(file_get_contents($fa_icons));
+  cache_set('stanford_icon_grid_icons', $fa_icons);
+  return $fa_icons;
+}
+
+/**
+ * Implements hook_field_formatter_info().
+ */
+function stanford_icon_grid_field_formatter_info() {
+  return array(
+    'fontawesome_icon_formatter' => array(
+      'label' => t('Font Awesome Icons'),
+      'field types' => array('fontawesome_icon'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_field_formatter_view().
+ */
+function stanford_icon_grid_field_formatter_view($entity_type, $entity, $field, $instance, $langcode, $items, $display) {
+  $element = array();
+  foreach ($items as $delta => $item) {
+    if (isset($item['icon'])) {
+
+      // Builds font awesome tag <i class="fa fa-{value}" aria-hidden="true"/>.
+      $element[$delta] = array(
+        '#theme' => 'html_tag',
+        '#tag' => 'i',
+        '#value' => '',
+        '#attributes' => array(
+          'class' => array('fa', 'fa-' . $item['icon']),
+          'aria-hidden' => array("true"),
+        ),
+        '#attached' => array(
+          'css' => array(
+            drupal_get_path('module', 'stanford_icon_grid') . '/css/fontawesome/stanford_fa.css',
+            drupal_get_path('module', 'stanford_icon_grid') . '/css/stanford_icon_grid.css',
+          ),
+        ),
+      );
+    }
+  }
+  return $element;
+}
+
+/**
+ * Implements hook_preprocess_field().
+ */
+function stanford_icon_grid_preprocess_field(&$vars) {
+  if ($vars['element']['#field_name'] == 'field_s_icon_ext_columns') {
+    $offset = 0;
+
+    // If not a multiple of 3, set the offset to center the field item.
+    if (count($vars['items']) % 3) {
+      $offset = -2 * (count($vars['items']) % 3 - 3);
+    }
+
+    foreach ($vars['items'] as $delta => &$item) {
+      $item['#attributes']['class'][] = 'span4';
+
+      // Sets the offset to the first item of the last group of 3. Ex:
+      // 1 or 2 items - the very first will get the offset. 4 or 5 - the 4th
+      // item will get the offset and so on.
+      if ($offset && !($delta % 3) && count($vars['items']) - $delta < 3) {
+        $item['#attributes']['class'][] = "offset$offset";
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_entity_view().
+ */
+function stanford_icon_grid_entity_view($entity, $type, $view_mode, $langcode) {
+  if ($type == 'field_collection_item' && $entity->field_name == 'field_s_icon_ext_columns') {
```

Comment:
I like how you think. i do that alot too. just a habit for building out the first version.